docs/walkthroughs/envoy-connection-draining-on-shutdown.md

```diff
@@ -0,0 +1,165 @@
+# **Connection draining on Pod shutdown**
+
+When a container is terminated in kubernetes, by default a SIGTERM signal is sent to the PID 1 of the container to indicate that the process must stop. Envoy, by default, immediatly closes all connections and exits upon receiving a SIGTERM, which can cause errors for the connections that are in flight. To avoid this MARIN3R provides a mechanism to perform connection draining before shutting down the Envoy process.
```

Comment:
```suggestion
When a container is terminated in Kubernetes, by default a SIGTERM signal is sent to the PID 1 of the container to indicate that the process must stop. Envoy, by default, immediately closes all connections and exits upon receiving a SIGTERM, which can cause errors for the connections that are in flight. To avoid this MARIN3R provides a mechanism to perform connection draining before shutting down the Envoy process.
```